lib/services/sf.js

```diff
@@ -1214,6 +1232,16 @@ StateConnecting.prototype.continue = function ()
       // update all token-related information
       parent.tokenInfo.update(body.data);
 
+      if (connectionConfig.getClientStoreTemporaryCredential()) {
+        if(body.data.idToken){
+          CredentialManager.writeCredential(connectionConfig.account, body.data.displayUserName, 
```

Comment:
We should pass username from `connectionConfig`, because it's the actual source of the username that we use when we are reading the credential. We can't rely on `displayUserName` being the same as config username, because it's a field in Snowflake that can be set to whatever the customer wants.

Reply:
Okay, I will fix them. Then, the driver will not save the token if users do not provide the username. Could you please confirm that this is the correct design?

Reply:
Fixed